model/registry/db.yaml

```diff
@@ -533,3 +530,66 @@ groups:
           reference the [elasticsearch schema](https://raw.githubusercontent.com/elastic/elasticsearch-specification/main/output/schema/schema.json)
           in order to map the path part values to their names.
         examples: ['db.elasticsearch.path_parts.index=test-index', 'db.elasticsearch.path_parts.doc_id=123']
+  - id: registry.db.vector
+    prefix: db.vector
+    type: attribute_group
+    brief: >
+      This group defines attributes for vector databases.
+    attributes:
+      - id: similarity
+        type:
+          members:
+            - id: cosine
+              value: 'cosine'
+              brief: >
+                The cosine metric.
+              stability: experimental
+            - id: dot
+              value: 'dot'
+              brief: >
+                The dot product metric.
+              stability: experimental
+            - id: euclidean
+              value: 'euclidean'
+              brief: >
+                The euclidean distance metric.
+              stability: experimental    
+            - id: manhattan
+              value: 'manhattan'
+              brief: >
+                The Manhattan distance metric.
+              stability: experimental
+        stability: experimental
+        brief: >
+          The metric used in similarity search.
+        examples: 'cosine'
+      - id: id
+        type: string
+        stability: experimental
+        brief: >
+          The ID of vector.
+        examples: '5c56c793-69f3-4fbf-87e6-c4bf54c28c26'
+      - id: name
+        type: string
+        stability: experimental
+        brief: >
+          The name field as of the vector (e.g. a field name).
+        examples: 'vector'
+      - id: dimension
+        type: int
+        stability: experimental
+        brief: >
+          The dimension of the vector.
+        examples: [3]
+      - id: model
+        type: string
+        stability: experimental
+        brief: >
+          The model used for the embedding.
+        examples: 'text-embedding-3-small'
+      - id: query.top_k
```

Comment:
I think we should come up with a more common attribute not specific to vector dbs. 
Many databases allow to limit number of returned rows:
- JDBC has [Statement.setMaxRows](https://docs.oracle.com/javase/7/docs/api/java/sql/Statement.html#setMaxRows(int)), 
- Mongo allows to set a [limit](https://www.mongodb.com/docs/drivers/java/sync/current/fundamentals/crud/read-operations/limit/#specify-a-limit)

Suggesting `db.query.max_returned_items` or similar

Reply:
@lmolkova I see the similarity here but I think the `db.vector.query.top-k` is more specific from a semantic point of view and more related to vectors, since it specifies the top k results in order, starting from the most similar. In semantic search we have this **similarity** value that is always present in any result that we don't have in standard database. The `limit` parameter of SQL returns the first k results but not in order, it depends on how you build the query (e.g. using `ORDER BY`).
 I personally think we should keep `db.vector.query.top-k` and potentially add a `db.query.limit` (or `db.query.max_returned_items` as you suggested) in a separate PR.

Reply:
Out of the few dbs I checked, they use `limit` in vector search
- pgvector uses [traditional `limit`](https://github.com/pgvector/pgvector?tab=readme-ov-file#getting-started) - same with cosmos and other sql databases
- mongo uses [imit](https://www.mongodb.com/docs/atlas/atlas-vector-search/tutorials/vector-search-quick-start/#std-label-vector-search-quick-start)
- qdrant uses [limit](https://qdrant.tech/documentation/concepts/search/#search-api)


So we're saying that DB instrumentations will need to detect if query is related to vector search or not and depending on this populate top-k or limit. That's difficult or impossible, but most importantly inconsistent and depends on instrumentation capabilities.

I.e. instrumentations that don't have vector-db specifics and those that do will use different attributes for the same thing.

So, I'd still prefer `db.query.limit` or something similar (and it should be under the same condition as `db.query.text` - we cannot require instrumentations to do query parsing)

Reply:
@lmolkova do you agree that `top-k` and `limit` are two different concepts, based on my previous comment? If they are I think we cannot use a single attribute (e.g. `db.query.limit`) to manage both.

Reply:
@lmolkova just a reminder for this, thanks.

Reply:
@ezimuel I see that databases use both terms to describe the same thing (see my comment above).

Let's say you have a postgres query like `SELECT * FROM items ORDER BY embedding <-> '[3,1,2]' LIMIT 5;` - the general-purpose DB instrumentation can report `limit`. If you make it understand vector search syntax, it may be able to use `top_k` instead, but that's would be inconsistent and unfamiliar for those who use vector search in postgres.

Reply:
@lmolkova I see your point but I think `top-k` has a different meaning from `limit`. If you are using a relation database as vector db, `limit` is fine since you are building an SQL statement and you specify the order. But, if you are using a native vector database (e.g. Qdrant), the `top-k` is more relevant since `top` implies the order, using a similarity metric.

I think we should add both:
- `db.query.limit`
- `db.vector.query.top-k`

Reply:
@lmolkova I found that the `SELECT LIMIT` is not part of the SQL standard. A statement compliant with SQL standard is `FETCH FIRST`. Moreover, I discovered that we already have `db.operation.parameter,<key>` in the semantic convention. This means, we can have `db.operation.parameter.fetch-first`. I don't think we need to add a `limit` or `fetch-first` in `db.query`. What do you think?

At the same time, I believe we should include `db.vector.query.top-k`. As noted earlier, it differs in meaning from `fetch-first` and, additionally, this parameter is much easier to retrieve in a vector database because it's not part of any statement.